### PR TITLE
EOS-17458 : CortxV2 SSPL: Enhance System Manifest

### PR DESCRIPTION
**Pull request by satish-seagate**
-- Adding Resource Map in Cortx Utils

Signed-off-by: Satish Darade <satish.darade@seagate.com>

# Problem Statement
- Improvement code.

# Design
-  This framework code will be used across all components when required. It is a generic common code.

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: 
-  Confirm All CODACY errors are resolved [Y/N]: 

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [x] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [x] All the tests performed should be mentioned before Resolving a JIRA. 
- [x] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/utils/discovery/resource_map.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    """Abstract class for all other resource types."""
+
+    name = "resource_map"
+
+    def __init__(self):
+        """Initialize resource."""
+        pass
+
+    @abstractmethod
+    def get_health_info(self, rpath):
```
Comment (ujjwalpl) [inline py-utils/src/utils/discovery/resource_map.py:31]: def get_health_info(self, rpath: str):

Also add return value.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/discovery/resource_map.py (hunk 1 above)
Comment (satish-seagate) [reply py-utils/src/utils/discovery/resource_map.py]: Done

**Review by ujjwalpl — APPROVED** [review]

File: py-utils/src/utils/discovery/resource_map.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    name = "resource_map"
+
+    def __init__(self):
+        """Initialize resource."""
+        pass
+
+    @abstractmethod
+    def get_health_info(self, rpath: str):
+        """
+        Get health information of fru in given resource id.
```
Comment (cdeshmukh) [inline py-utils/src/utils/discovery/resource_map.py:33]: Get health information of resource in given rpath.

   :param str rpath: Resource path (Example: hw>disks)
   :return: resource manifest as dictionary or list of dictionary, based on input rpath

File: py-utils/src/utils/discovery/resource_map.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+        :param str rpath: Resource id (Example: hw>disks)
+        :return: list of dictionary or dictionary based on input rpath
+        """
+        pass
+
+    @abstractmethod
+    def get_manifest_info(self, rpath: str):
+        """
+        Get manifest information of fru in given resource id.
```
Comment (cdeshmukh) [inline py-utils/src/utils/discovery/resource_map.py:43]: Get manifest information of resource in given rpath.

   :param str rpath: Resource path (Example: hw>disks)
   :return: resource manifest as dictionary or list of dictionary, based on input rpath

File: py-utils/src/utils/discovery/resource_map.py (hunk 2 above)
Comment (satish-seagate) [reply py-utils/src/utils/discovery/resource_map.py]: Done

File: py-utils/src/utils/discovery/resource_map.py (hunk 3 above)
Comment (satish-seagate) [reply py-utils/src/utils/discovery/resource_map.py]: Done

**Review by cdeshmukh — APPROVED** [review]

**Comment by sachinpunadikar** [comment]
@satish-seagate please fill up the PR template

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**